### PR TITLE
10831 link geojson featureid

### LINKED ISSUE #10831: include database link between geoms in geojson_geometries and individual features in a tile.data feature collection
This issue is essentially about the fact that each geospatial feature exists independent of the tile and resource it's associated with because there can be N features per tile. Basically if you're looking for a particular geom via its `featureid`, you can't find it via the database, only ES. If a given tile has N geoms, the db doesn't know one from the other except via that integer primary key. I would characterize a solution to this as "Nice to Have" as opposed to "necessary" because the `"id"` property on the feature is part of the ES `terms` index, so each individual feature is searchable if you know that id. [This change](https://github.com/archesproject/arches/pull/10827/commits/40959570cb3f87a330ecc461405809f64ab1b3b2) 
 included in PR #10827 also ensures that when we create new tiles with known feature ids they don't get overwritten.

### PR DESCRIPTION
**Pull request by whatisgalen**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

- creates a new field on the model `GeoJSONGeometry` called `featureid` that points to the `"id"` property on geojson features in tiledata
- creates a migration to add this field and update both the `refresh_geojson_geometries()` and `refresh_tile_geojson_geometries()` postgres functions to save the `"id"` property from tile data features into `featureid` for corresponding rows in `geojson_geometries`
- includes `featureid` as property on MVT called `tile_featureid`

### Issues Solved

Closes #10831 

### Checklist

-   I targeted one of these branches:
    - [x] dev/7.6.x (under development): features, bugfixes not covered below
    - [ ] dev/7.5.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [ ] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [x] Unit tests pass locally with my changes
-   [ ] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Accessibility Checklist

[Developer Guide](https://arches.readthedocs.io/en/stable/developing/advanced/accessibility/)

| Topic            | Changed | Retested |
| ---------------- | ------- | -------- |
| Color contrast   |         |          |
| Form fields      |         |          |
| Headings         |         |          |
| Links            |         |          |
| Keyboard         |         |          |
| Responsive Design|         |          |
| HTML validation  |         |          |
| Screen reader    |         |          |

#### Ticket Background
*   Sponsored by: @scholiumtech
*   Found by: @whatisgalen 
*   Tested by: @robgaston 
*   Designed by: @whatisgalen

### CODE REVIEW

File: arches/app/models/migrations/10799_geojsongeometry_featureid.py
Comment (robgaston) [inline arches/app/models/migrations/10799_geojsongeometry_featureid.py:1]: @whatisgalen I notice the where clauses have changed in the functions you edited here, but I don't understand why that should be necessary for this change.  It seems to me like we should only be adding the one new column (`featureid`) and not doing anything else.  Am I missing something?

File: arches/app/models/migrations/10799_geojsongeometry_featureid.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                                )
+                            )).geom,
+                            4326
+                        ),
+                        3857
+                    )
+                ) AS geom,
+                (json_array_elements(
+                    t.tiledata::json->n.nodeid::text->'features'
+                )::json->>'id')::uuid AS featureid
```
Comment (robgaston) [inline arches/app/models/migrations/10799_geojsongeometry_featureid.py:46]: the cast to `json` here is unnecessary, I think, as `json_array_elements` returns `json`

File: arches/app/views/api.py [hunk 2]
```diff
@@ -455,8 +455,10 @@ def get(self, request, nodeid, zoom, x, y):
                         cursor.execute(
                             """SELECT ST_AsMVT(tile, %s, 4096, 'geom', 'id') FROM (SELECT tileid,
                                 id,
+                                featureid AS tile_featureid,
                                 resourceinstanceid,
                                 nodeid,
+                                featureid::text AS tile_featureid,
```
Comment (robgaston) [inline arches/app/views/api.py:461]: you seem to have the column duplicated in this select statement.  Also, is there a reason to use the name `tile_featureid` over the simpler and more consistent `featureid`?

**Review by robgaston — CHANGES_REQUESTED** [review]

File: arches/app/models/migrations/10799_geojsongeometry_featureid.py
Comment (whatisgalen) [reply arches/app/models/migrations/10799_geojsongeometry_featureid.py:1]: I can't remember why I added the GROUP BY clause but I've removed it and reformatted the lines so the where clauses are unchanged

File: arches/app/models/migrations/10799_geojsongeometry_featureid.py (hunk 1 above)
Comment (whatisgalen) [reply arches/app/models/migrations/10799_geojsongeometry_featureid.py]: change in latest

File: arches/app/views/api.py (hunk 2 above)
Comment (whatisgalen) [reply arches/app/views/api.py]: change in latest

**Review by robgaston — APPROVED** [review]

**Pull request merged**